ui/src/plugins/fetch.js

```diff
@@ -41,6 +41,12 @@ export function fetchWithContext(
     });
 }
 
+/**
+ * @param {string} path 
+ * @returns path with '/' not duplicated except at ://
+ * 
+ * Note: we need the second .replace() because the first one doesn't catch multiple slashes at the start of the string
+ */
 export function cleanDuplicateSlash(path) {
-  return path.replace(/([^:]\/)\/+/g, "$1");
+  return path.replace(/([^:]\/)\/+/g, "$1").replace(/^(\/)+/, '/');
```

Comment:
Why do we need to replace twice?
I think we should find another regex for this.

Reply:
The second replace isn't a costly operation as the associated regex has `^`.

Having two regex also ensures the code is more readable with simpler regexes...

Reply:
@haricane8133 Happy new year!
Sorry for late replying.
Here is the regex that I found: `/(https?:\/\/)|(\/)+/g`
Ex: `path.replace(/(https?:\/\/)|(\/)+/g, "$1$2")`

![image](https://github.com/conductor-oss/conductor/assets/83625759/6df0ade2-ec1d-4b95-9342-4df52aed8b7e)

Reply:
@nhandt2021, this is clever.

But it currently fails for
`'https:///some-domain.com//sub-domain/'.replace(/(https?:\/\/)|(\/)+/g, "$1$2")`

Reply:
I would just add a `\/*` after `(https?:\/\/)` to fix it

`'https:///some-domain.com//sub-domain/'.replace(/(https?:\/\/)\/*|(\/)+/g, "$1$2")`

Do you think it would be better not to pin the protocol? The Util may be used else-where too...

How about this? We can remove `https?` from the regex
`'https:///some-domain.com//sub-domain/'.replace(/(:\/\/)\/*|(\/)+/g, "$1$2")`

Reply:
I have pushed the code with the latest regex, and rebased commits from upstream.

Reply:
Actually I don't think we need to cover this case because it's completely wrong syntax.
How about if user input a URL like this: `https:#$@@#$@*&#@////////some-domain.abc`

Reply:
We aren't solving for 100% of the possibilities...
(we have a `+ '/'` in our code base elsewhere and we cannot be sure where that extra '/' would be appended)

But we **MUST** make sure that the function, which deals only with '/', does what is expected out of it.

Reply:
> Do you think it would be better not to pin the protocol? The Util can be used elsewhere too... (the existing code didn't specify the protocol)
> 
> How about this? We can remove `https?` from the regex `'https:///some-domain.com//sub-domain/'.replace(/(:\/\/)\/*|(\/)+/g, "$1$2")`
> 
> Let us have the function as generic as possible

Agree with this :)